test/framework/fixtures.go

```diff
@@ -0,0 +1,104 @@
+package framework
+
+import (
+	"github.com/k8ssandra/k8ssandra-operator/pkg/utils"
+	"path/filepath"
+	"strings"
+)
+
+// TestFixture is a set of one or more yaml manifests, typically a manifest for a K8ssandraCluster. They are
+// automatically deployed right before the test itself is executed.
+type TestFixture struct {
+
+	// Name specifies the name of the fixture. The fixture name resolves to a subdirectory under the
+	// test/testdata/fixtures directory.
+	Name string
+}
+
+func NewTestFixture(name string) *TestFixture {
+	return &TestFixture{Name: name}
+}
+
+var (
+	fixturesSrcDir  = filepath.Join("..", "testdata", "fixtures")
+	fixturesDestDir = filepath.Join("..", "..", "build", "test-config", "fixtures")
+)
+
+func (t *TestFixture) fixtureDir() string {
+	path := filepath.Join(fixturesSrcDir, t.Name)
+	dir, err := filepath.Abs(path)
+	if err != nil {
+		panic(err)
+	}
+	return dir
+}
+
+func (f *E2eFramework) DeployFixture(namespace string, fixture *TestFixture) error {
+	numDcContexts, err := getNumDcContexts(fixture)
+	if err != nil {
+		return err
+	}
+	dcContexts := f.AllK8sContexts()[:numDcContexts]
+	err = generateFixtureKustomizationFile(namespace, fixture.Name, dcContexts)
+	if err != nil {
+		return err
+	}
+	destDir := filepath.Join(fixturesDestDir, fixture.Name)
+	return f.kustomizeAndApply(destDir, namespace, f.ControlPlaneContext)
+}
+
+// getNumDcContexts computes the exact number of contexts to kustomize by counting the number of dcs/contexts in the
+// K8ssandraCluster spec for this fixture, if any. This is a bit hacky but unfortunately kustomize errors out if we
+// attempt to patch a non-existing path.
+func getNumDcContexts(fixture *TestFixture) (int, error) {
+	files, err := utils.ListFiles(fixture.fixtureDir(), "k8ssandra.yaml")
```

Comment:
Would it be simpler to use a yaml parse to do this?

Reply:
I think the best tool for this would be yq. The following gives you the total number of `k8sContext` fields declared in the spec:

```
yq ea '.spec.cassandra.datacenters.[] | .k8sContext as $item ireduce (0; . +1)' k8ssandra.yaml
```

Reply:
The following is even better as it would give me which dcs have a k8sContext defined:

```
 > yq e '.spec.cassandra.datacenters.[] | .k8sContext ' k8ssandra.yaml 
null
kind-k8ssandra-1
```

With this I know that the kustomization must target only `.spec.cassandra.datacenters[1].k8sContext`.

I can create a ticket to implement that if you like.

Reply:
That would be great!

Reply:
done, see #462